app/forms/AssistanceDetailsForm.scala

```diff
@@ -17,41 +17,69 @@
 package forms
 
 import connectors.exchange.AssistanceDetails
-import play.api.data.Form
+import play.api.data.{ FormError, Form }
 import play.api.data.Forms.{mapping, of, optional}
+import play.api.data.format.Formatter
 
 object AssistanceDetailsForm {
 
+  private[forms] val deptMaxLength = 256
+
+  // scalastyle:off cyclomatic.complexity
+  val disabilityAndGisDependentFormatter = new Formatter[Option[String]] {
+    override def bind(key: String, data: Map[String, String]): Either[Seq[FormError], Option[String]] = {
+      val disabilityCheck = data.get("hasDisability")
+      val gisCheck = data.get("guaranteedInterview")
+      val value = data.get(key).filterNot(_.trim.isEmpty) // value for needsSupportForOnlineAssessment
+
+      (disabilityCheck, gisCheck, value) match {
```

Comment:
Maybe, it can be simplify a bit by adding few private functions, and use only one-level pattern matching, like this:
```
disabilityCheck match {
   case Some("Yes") => validateWhenYes(...)
   case Some("No") => validateWhenNo(...)
   case Some("I don't know...") => validateWhenUnknown(...)
}
```